src/RobotSimulator.elm

```diff
@@ -0,0 +1,305 @@
+module Main exposing (main, update, view)
+
+import Browser
+import Browser.Events
+import Html exposing (..)
+import Html.Attributes exposing (style)
+import Html.Events exposing (onClick)
+import Json.Decode as Decode
+import Time
+
+
+screenWidth =
+    500
+
+
+screenHeight =
+    500
+
+
+shotVelocity =
+    Vector 1 1
+
+
+main =
+    Browser.element
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+type alias Vector =
+    { x : Float, y : Float }
+
+
+type alias GameObject =
+    { position : Vector, velocity : Vector, direction : Direction }
+
+
+type alias Model =
+    { robot : GameObject, shots : List GameObject }
+
+
+type Direction
+    = Right
+    | Left
+    | Up
+    | Down
+
+
+type Msg
+    = Move Direction
+    | Shoot
+    | Interval
+
+
+addVectors : Vector -> Vector -> Vector
+addVectors baseVector addedVector =
+    Vector (baseVector.x + addedVector.x) (baseVector.y + addedVector.y)
+
+
+asShotsIn : Model -> List GameObject -> Model
+asShotsIn model newShots =
+    { model | shots = newShots }
+
+
+asRobotIn : Model -> GameObject -> Model
+asRobotIn model newRobot =
+    { model | robot = newRobot }
+
+
+asPositionIn : GameObject -> Vector -> GameObject
+asPositionIn gameObject newPosition =
+    { gameObject | position = newPosition }
+
+
+asDirectionIn : GameObject -> Direction -> GameObject
+asDirectionIn gameObject newDirection =
+    { gameObject | direction = newDirection }
+
+
+
+-- Init
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    let
+        robot =
+            GameObject { x = 50, y = 50 } { x = 5, y = 5 } Up
+
+        shots =
+            []
+    in
+    ( Model robot shots, Cmd.none )
```

Comment:
Instead of `init _ =`, just change the last line to `always ( Model robot shots, Cmd.none )`.
Either way, you don't need specifically `()` so the signature should be `init : a -> ( Model, Cmd Msg )`.